server/container_create.go

```diff
@@ -373,3 +383,20 @@ func (s *Server) generateContainerIDandName(podName string, name string, attempt
 	}
 	return id, name, err
 }
+
+// getAppArmorProfileName gets the profile name for the given container.
+func (s *Server) getAppArmorProfileName(annotations map[string]string, ctrName string) string {
+	profile := apparmor.GetProfileNameFromPodAnnotations(annotations, ctrName)
+
+	if profile == "" {
+		return ""
+	}
+
+	if profile == apparmor.ProfileRuntimeDefault {
+		// If the value is runtime/default, then return default profile.
+		return s.appArmorProfile
+	}
+
+	profileName := strings.TrimPrefix(profile, apparmor.ProfileNamePrefix)
```

Comment:
we could get rid of this var altogether by just returning the assignment, your call though.